src/UnigramWordPredictor.java

```diff
@@ -51,7 +55,19 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
-    // TODO: Convert the trainingWords into neighborMap here
+    //Iterating throught the tokens
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+      //Saving the current word of the for loop into currentString
+      String currentString = trainingWords.get(i);
+      //Saving the next word of the for loop into nextString
+      String nextString = trainingWords.get(i + 1);
+      //Checks if the current word is in the neighborMap
+      if (!neighborMap.containsKey(currentString)) {
+        neighborMap.put(currentString, new ArrayList<>());
+      }
+      //Adds the word to the map
+      neighborMap.get(currentString).add(nextString);
+    }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?